state/accumulator.go

```diff
@@ -291,131 +292,125 @@ func (a *Accumulator) Initialise(roomID string, state []json.RawMessage) (Initia
 //     to exist in the database, and the sync stream is already linearised for us.
 //   - Else it creates a new room state snapshot if the timeline contains state events (as this now represents the current state)
 //   - It adds entries to the membership log for membership events.
-func (a *Accumulator) Accumulate(roomID string, prevBatch string, timeline []json.RawMessage) (numNew int, timelineNIDs []int64, err error) {
-	if len(timeline) == 0 {
-		return 0, nil, nil
-	}
```

Comment:
Is it safe to remove this? If the timeline is empty, then I think we'll try to insert 0 rows into the events table. Not sure if that will cause an error?

Reply:
This check was moved up into Storage.Accumulate.